### ISSUE TITLE
PyNvVideoCodec inference dataloader as an alternative to DALI

### ISSUE THREAD

**Issue by themattinthehatt** [danbider/lightning-pose#476]
## Motivation

`litpose predict` on video currently always routes through DALI (`PrepareDALI` / `LitDaliWrapper` in `lightning_pose/data/dali.py`). For inference specifically, DALI's randomized/multiview-sync/imgaug-augmentation machinery is unused — `predict` already runs with shuffle and imgaug off. A decoder built on [PyNvVideoCodec](https://developer.nvidia.com/pynvvideocodec) (NVDEC-backed, random-access-by-index) can replicate the much smaller subset actually needed for inference — sequential (optionally multiview-synced) frame reading, context-model overlapping windows, GPU resize+normalize, and bbox-crop — while giving simpler control over context-window construction than DALI's `step`/`pad_sequences` machinery.

This should be a drop-in alternate backend selected at predict time, fully compatible with the existing `--runtime`/`--compile` (eager / torch.compile / ONNX) axis, since it only replaces video ingestion, not the model or `Trainer.predict()` call.

## Scope

`litpose predict` on video inputs only. Image and CSV prediction (`predict_on_label_csv`) use a plain `DataModule` and never touch DALI/video decoding — out of scope.

## CLI

- New flag: **`--decoder`**, choices `dali` / `pynvvc`.
- Default: auto-select `pynvvc` if available (package installed **and** GPU/driver actually support NVDEC — see "Install & default-selection guard" below), else fall back to `dali`. Flag stays optional.
- If the user explicitly passes `--decoder pynvvc` and it isn't usable, fail fast in the CLI handler with an actionable message (mirrors the existing `--compile` + `--runtime onnx` guard in `predict.py:183`), rather than surfacing a raw NVIDIA exception.
- Print which decoder backend is in use. Put this print in `predict_video()` (`lightning_pose/utils/predictions.py`) rather than duplicating it in the CLI and Model API, so every caller (CLI, direct `Model` API/notebook use) gets it for free.

## Model API

- New per-call kwarg on `Model.predict_on_video_file` and `Model.predict_on_video_file_multiview`.
- Threaded through to `predict_video()` in `lightning_pose/utils/predictions.py`, same as `bbox_file` already is.

## Branch point

`predict_video()` (`lightning_pose/utils/predictions.py:412`) is the single place `PrepareDALI` is constructed today. Add a branch there: `dali` → existing `PrepareDALI`/`vid_pred_class()` path unchanged; `pynvvc` → new path. `PredictionHandler` and the `trainer.predict(...)` call do not change.

## New file: `lightning_pose/data/pynvvc.py`

Mirrors `dali.py`'s two-phase construction (`PrepareDALI.__init__` validates/precomputes, `__call__` builds the iterator). Needs to produce `UnlabeledBatchDict` / `MultiviewUnlabeledBatchDict` exactly as consumed by `predict_step` in `heatmap_tracker.py`/`heatmap_tracker_mhcrnn.py`/`heatmap_tracker_multiview.py`/`regression_tracker.py`:

- `frames`: FCHW layout, `[0,1]`-scaled then ImageNet-normalized.
- `transforms`: for prediction (no `dlc`/`dlc-top-down` imgaug), this is just the sentinel `np.array([-1])` per `dali.py:180-182` — no rotation/scale matrix to reconstruct. Augmentation-transform replication only matters for training, which is out of scope.
- `bbox`: full-frame bbox by default, or actual per-frame bbox in crop mode.
- Context models: replicate the 5-frame overlapping-window behavior and the "pad last batch by repeating the final frame" behavior so shapes stay static across steps (see torch.compile note below) and `PredictionHandler` still gets full tensors matched to the true frame count.

## Extract bbox-crop logic into `lightning_pose/data/bboxes.py`

`_apply_bbox_crop` (`dali.py:332-396`) is currently a method coupled to `LitDaliWrapper`'s `self.bbox_df`/`self.do_context`/`self.resize_dims`/`self._frame_idx`. To avoid forking ~65 lines of crop/clamp/resize logic (including the edge-clamping fix for negative x/y from `create_bbox`) into a second class, pull the per-frame crop+resize math into a free function in `bboxes.py` — which is already the module for bbox coordinate-space transforms (`frame_to_model`, `model_to_frame_batch`, etc.):

```python
def crop_and_resize_frames(
    frames: Float[torch.Tensor, "seq_len rgb h w"],
    bbox_rows: pd.DataFrame,
    resize_dims: list[int],
) -> tuple[Float[torch.Tensor, "seq_len rgb h_out w_out"], Float[torch.Tensor, "seq_len xyhw"]]:
```

Takes pre-sliced/padded bbox rows; leaves cursor/step bookkeeping (`_frame_idx`, `seq_len - 4` context step for DALI vs. pynvvc's windowing) in each backend's wrapper. Both `LitDaliWrapper._apply_bbox_crop` and the new pynvvc wrapper call this shared function.

## Install & default-selection guard

- Add `pynvvideocodec` to `pyproject.toml`'s unconditional `dependencies` list (same treatment as `nvidia-dali-cuda110` at line 54), gated by the same `sys_platform == 'linux' and platform_machine == 'x86_64'` marker. Package is ~245 KB compressed — no install-size concern.
- GPU/driver reality: NVDEC required (Turing/Ampere/Ada/Hopper/Blackwell only — Pascal and older unsupported), Linux driver ≥ 530.41.03. `import pynvvideocodec` will succeed regardless of GPU generation or driver age — the failure only surfaces when a decoder is actually constructed. **`is_pynvvc_available()` must attempt a minimal decoder construction inside its try/except, not just the import**, or auto-selection will pick `pynvvc` on an unsupported machine and crash on the first video with a raw NVIDIA error instead of falling back to DALI. One-time cost per predict call, not per-batch — cheap enough to do properly.
- Follow the CPU-only-install lazy-import discipline already documented in CLAUDE.md for `dali.py`: `pynvvideocodec` import stays inside function bodies in `pynvvc.py`, never at module top level, marked with the same `# lazy: avoids ImportError on cpu-only installs` convention.

## CUDA stream synchronization — real risk, needs explicit testing

PyNvVideoCodec hands back decoded frames as DLPack buffers (`torch.from_dlpack(frame)`), decoded asynchronously on the decoder's own CUDA stream. If the model's forward pass (default stream, or a `torch.compile`/cudagraphs side-stream) consumes that tensor without the consumer stream waiting on the decode-complete event, the buffer can be read before NVDEC finishes writing it. This fails silently — no crash, no NaN, just quietly wrong pixels feeding a plausible-looking keypoint prediction.

- Check first whether the decoder can be constructed against an explicit external stream handle (pass `torch.cuda.current_stream()`'s stream in) — if supported, that's strictly better than per-batch `synchronize()` calls: no sync cost, no risk of a missed call on a new code path.
- Test plan: standalone repro script, decode → `torch.from_dlpack` → run under each of the three compute backends (eager, `torch.compile`, ONNX Runtime CUDA EP) separately, since stream handling differs per backend and a sync bug can be backend-specific (cudagraphs capture is especially stream-order-sensitive). Run predictions twice — once with an explicit sync/stream-wait inserted right after the DLPack conversion, once without — and diff the output keypoint CSVs. Also test under concurrent GPU load (two `predict_video` calls at once, or decoding while `--compile` is still warming up) — divergence that only appears under load is the race; a small consistent pixel-error offset between pynvvc and DALI outputs is expected and fine (different resize kernels, not a bug).

## torch.compile / ONNX / TensorRT compatibility

Should be automatic since `predict_step` and the `trainer.predict()` call site are unchanged. One thing the pynvvc wrapper must get right: DALI already pads the last batch to full `seq_len` for context models so shapes stay static across steps; the new "pad by repeating final frame" logic must produce identical shapes, or torch.compile will eat a recompilation stall on the final batch of every video.

## Docs

Make explicit in the inference-speed doc page that `--decoder` (video ingestion: dali/pynvvc) and `--runtime`/`--compile` (model execution: eager/onnx/compiled) are independent axes — show at least one example combining them (`--decoder pynvvc --runtime onnx`), plus a short table of valid combinations (all valid except the existing `--compile` + `--runtime onnx` restriction).

## Files touched

- [ ] `lightning_pose/cli/commands/predict.py` — `--decoder` flag
- [ ] `lightning_pose/api/model.py` — new kwarg on `predict_on_video_file`/`_multiview`
- [ ] `lightning_pose/utils/predictions.py` — branch in `predict_video()`, backend print
- [ ] `lightning_pose/data/pynvvc.py` — new wrapper + `is_pynvvc_available()`
- [ ] `lightning_pose/data/bboxes.py` — extracted `crop_and_resize_frames`
- [ ] `lightning_pose/data/dali.py` — `LitDaliWrapper._apply_bbox_crop` calls the shared helper
- [ ] `pyproject.toml` — unconditional dependency, platform-gated
- [ ] `tests/data/test_pynvvc.py` — mirrors `tests/data/test_dali.py`, gpu-marked
- [ ] inference-speed doc page — `--decoder` flag, independence from `--runtime`/`--compile`
